pkg/ddc/juicefs/sync_runtime.go

```diff
@@ -253,20 +253,20 @@ func (j *JuiceFSEngine) isVolumeMountsChanged(crtVolumeMounts, runtimeVolumeMoun
 }
 
 func (j JuiceFSEngine) isEnvsChanged(crtEnvs, runtimeEnvs []corev1.EnvVar) (changed bool, newEnvs []corev1.EnvVar) {
-	envMap := make(map[string]string)
+	envMap := make(map[string]corev1.EnvVar)
 	for _, env := range crtEnvs {
-		envMap[env.Name] = env.Value
+		envMap[env.Name] = env
 	}
 	for _, env := range runtimeEnvs {
-		if envMap[env.Name] != env.Value {
+		if envMap[env.Name].Value != env.Value || envMap[env.Name].ValueFrom != env.ValueFrom {
```

Comment:
ValueFrom is a pointer, so `envMap[env.Name].ValueFrom != env.ValueFrom` may not work. How about using `if !reflect.deepEqual(envMap[env.Name], env)`?

Reply:
done